### PR TITLE
Minor corrections on resilience page

### PR DESCRIPTION
**Pull request by peter-csala**
## Summary

- Removed the circuit breaker from the example
  - Added options to make the retry timeout-aware
  - Updated the `key` accordingly 
- Added `AddConcurrencyLimiter` next to `AddRateLimiter`
- Corrected `AddHedging` description
  -  The original description described the _parallel_ mode, but [it can work](https://www.pollydocs.org/strategies/hedging.html#concurrency-modes) in _latency_ and _fallback_ modes as well.
- Corrected the number of attempts in case of retry
- Replaced `await ValueTask.CompletedTask` to `return default` 

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/resilience/index.md](https://github.com/dotnet/docs/blob/c85748f088a0049e12dac2c6f6761ea819adbc51/docs/core/resilience/index.md) | [Introduction to resilient app development](https://review.learn.microsoft.com/en-us/dotnet/core/resilience/index?branch=pr-en-us-37869) |

### CODE REVIEW

**Comment by peter-csala** [comment]
@IEvangelist Hy Dave, can you please review it?

File: docs/core/resilience/snippets/resilience/Program.cs [hunk 1]
```diff
@@ -36,11 +37,11 @@
 // </pipeline>
 
 // <execute>
-await pipeline.ExecuteAsync(static async cancellationToken =>
+await pipeline.ExecuteAsync(static cancellationToken =>
 {
     // Code that could potentially fail.
 
-    await ValueTask.CompletedTask;
+    return default; // ValueTask.CompletedTask;
```
Comment (IEvangelist) [inline docs/core/resilience/snippets/resilience/Program.cs:44]: Why change this?

**Review by IEvangelist — APPROVED** [review]
Everything else looks good, except for the change to the execute call.

File: docs/core/resilience/snippets/resilience/Program.cs (hunk 1 above)
Comment (peter-csala) [reply docs/core/resilience/snippets/resilience/Program.cs:44]: return default; is sufficient to compile and does not require an async state machine.

Comment (peter-csala) [reply docs/core/resilience/snippets/resilience/Program.cs:44]: @IEvangelist Can you accept my rationale or shall I rollback this change to your original?

Comment (IEvangelist) [inline docs/core/resilience/snippets/resilience/Program.cs:44]:
```suggestion
    return ValueTask.CompletedTask;
```

**Review by IEvangelist — APPROVED** [review]
Looks good to me, we'll :shipit: - thank you 🙏

**Comment by peter-csala** [comment]
> > Shall I push a commit to change the csproj file?
> 
> Yes, please - thank you.

@IEvangelist Done :)

**Comment by IEvangelist** [comment]
Hey @peter-csala - I'll monitor this one and update it next week when .NET 8.0 ships, that way we can just use the stable release.

**Comment by peter-csala** [comment]
> Hey @peter-csala - I'll monitor this one and update it next week when .NET 8.0 ships, that way we can just use the stable release.

Thank you Sir ;)

**Pull request closed without merging**